tss-esapi/src/structures/buffers/private.rs

```diff
@@ -7,9 +7,36 @@ use crate::{
 };
 use std::convert::TryInto;
 use tss_esapi_sys::_PRIVATE;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
 
 buffer_type!(Private, ::std::mem::size_of::<_PRIVATE>(), TPM2B_PRIVATE);
 
+impl Serialize for Private {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let bytes = self.marshall()
```

Comment:
Given that this just relies on `Marshall` underneath, wouldn't it make sense to just implement these two traits generically for all types `T: Marhsall + UnMarshall`?

Reply:
I suspect that if we make it generic in that way then you break users who have their own generics relying on Serialise/Deserialise since then we'd have a Serialise  + Marshall bound?

Reply:
Since you can only implement traits for types defined in your crate, they'd only be able to implement it on their own types. I'm not really sure Marshall/UnMarshall make sense for types outside of our crates, tbh, and maybe we should document it as such. Also since this change will be going out in a major change, it won't break someone's code all of a sudden without an intentional update to their dependencies.

Reply:
Okay, so is there anything actionable here then? I think the way it is now is fine, we could always macro it to save boiler plate I guess?

Reply:
I think it can go in as is, we'll maybe add the generic implementation down the line.